src/components/Form/ExitFormDialogProvider.tsx

```diff
@@ -237,16 +246,27 @@ const ExitFormDialogProvider = ({ children }) => {
     shouldBlockNavigation,
     setEnableExitDialog,
     setExitDialogSubmitRef: setSubmitRef,
-    setIsSubmitting
+    setIsSubmitting,
+    setIsSubmitDisabled
   };
 
+  useBeforeUnload(e => {
+    // If form is dirty and user does a refresh,
+    // the browser will ask about unsaved changes
+    if (shouldBlockNav() && !isInDevelopment) {
```

Comment:
I don't understand why it should behave differently in dev and prod mode

Reply:
Mainly because it's annoying to develop forms with this feature. It stops page refresh on code change.